src/com/xilinx/rapidwright/edif/EDIFTools.java

```diff
@@ -532,9 +532,40 @@ public static void connectPortInstsThruHier(EDIFHierPortInst src, EDIFHierPortIn
             EDIFHierCellInst hierParentInst = hierPortInst.getHierarchicalInst();
             EDIFNet currNet = hierPortInst.getNet();
             if (currNet == null && !(hierParentInst.equals(commonAncestor) && hierPortInst == snk)) {
-                if (hierPortInst == src) createdSrcNet = true;
-                currNet = createUniqueNet(hierParentInst.getCellType(), newName);
-                currNet.addPortInst(hierPortInst.getPortInst());
+                // The snk pin is not connected to a net, let's start with the common ancestor
+                // and search downward
```

Comment:
```suggestion
                // When operating on the snk pin, we've found that it is not connected to a net.  Instead of trying to
                // connect through the hierarchy starting at the snk, start from the common ancestor and leverage
                // as much existing connectivity as possible.
```